src/main/java/org/reduct/client/ReductClient.java

```diff
@@ -54,4 +59,29 @@ <T> HttpResponse<String> sendRequest(String url, Method method) throws ReductExc
          throw new ReductSDKException("Thread has been interrupted while processing the request", e);
       }
    }
+
+   public <T> HttpResponse<T> send(HttpRequest.Builder builder, HttpResponse.BodyHandler<T> bodyHandler) {
+      try {
+         HttpResponse<T> httpResponse = getHttpClient().send(builder.build(), bodyHandler);
+         switch (httpResponse.statusCode()) {
+            case 200 -> {
+               return httpResponse;
+            }
+            case 400 -> throw new ReductException(ERROR_400, httpResponse.statusCode());
```

Comment:
Just use the server's message from `x-reduct-error` header.

Reply:
Fixed